Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -94,22 +94,22 @@ private void TrackSplitsCost(List<Split> splits)
       }
     }
 
-    async Task DoWork(Split split)
+    async Task DoWork(Split split, CancellationToken cancellationToken)
     {
       var checker = await split.parent.CheckerPool.FindCheckerFor(split.parent, split);
 
       try {
-        StartCheck(split, checker);
+        cancellationToken.ThrowIfCancellationRequested();
+        StartCheck(split, checker, cancellationToken);
         await split.ProverTask;
-        await ProcessResult(split);
+        await ProcessResult(split, cancellationToken);
       }
-      catch {
+      finally {
         split.ReleaseChecker();
-        throw;
```

Comment:
Have you tested this against a badly-behaved prover? I'm worried that this could undo the fix from #488.

Reply:
Since we don't catch the exception, we also don't have to rethrow it since it's thrown up automatically. Where does your concern come from?

Reply:
Ah, right, the change from `catch` to `finally` should avoid it. And there is a test that would be failing if this change did undo that fix. 👍